pkg/operator/ceph/object/notification/obc_label_controller_test.go

```diff
@@ -263,6 +292,146 @@ func TestCephBucketNotificationOBCLabelController(t *testing.T) {
+		resetValues()
+		deleteOBC, deleteOB := createOBResources(deleteBucketName)
+		deleteOBC.Spec.GenerateBucketName = deleteBucketName
+		deleteOBC.Spec.ObjectBucketName = deleteBucketName
+		deleteOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		req.NamespacedName.Name = deleteBucketName
+		objects := []runtime.Object{
+			cephCluster,
+			deleteOBC,
+			deleteOB,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 0, createNotificationCount)
+		assert.Equal(t, 1, deleteNotificationCount)
+	})
+
+	t.Run("provision OBC with multiple notification labels", func(t *testing.T) {
+		resetValues()
+		multipleCreateOBC, multipleCreateOB := createOBResources(multipleCreateBucketName)
+		multipleCreateOBC.Labels = setNotificationLabels([]string{testNotificationName + "-1", testNotificationName + "-2"})
+		multipleCreateOBC.Spec.ObjectBucketName = multipleCreateBucketName
+		multipleCreateOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		bucketNotification1 := createBucketNotification(testNotificationName + "-1")
+		bucketNotification2 := createBucketNotification(testNotificationName + "-2")
+		req.NamespacedName.Name = multipleCreateBucketName
+
+		objects := []runtime.Object{
+			cephCluster,
+			multipleCreateOBC,
+			multipleCreateOB,
+			bucketNotification1,
+			bucketNotification2,
+			bucketTopic,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 2, createNotificationCount)
+		assert.Equal(t, 0, deleteNotificationCount)
+	})
+
+	t.Run("delete multiple notifications from the obc", func(t *testing.T) {
+		resetValues()
+		multipleDeleteOBC, multipleDeleteOB := createOBResources(multipleDeleteBucketName)
+		multipleDeleteOBC.Spec.GenerateBucketName = multipleDeleteBucketName
+		multipleDeleteOBC.Spec.ObjectBucketName = multipleDeleteBucketName
+		multipleDeleteOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		req.NamespacedName.Name = multipleDeleteBucketName
+		objects := []runtime.Object{
+			cephCluster,
+			multipleDeleteOBC,
+			multipleDeleteOB,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 0, createNotificationCount)
+		assert.Equal(t, 2, deleteNotificationCount)
```

Comment:
For creating and deleting notifications, how can we verify that the **correct** notifications are created and deleted? An important part of connecting this logic to the create/delete logic is that we should make sure what we are creating/deleting is what should be created/deleted. 

I would suggest using a slice of names/ids of notifications to ensure the logic is applying the correct names to the called functions.

Reply:
@BlaineEXE : I am kind of doing reverse logic here [createNotificationFunc](https://github.com/rook/rook/pull/9365/files#diff-39475f40154dad6ffbbe10347e21b251052adaf46f597105ad37cd8445ad65cfR77) and [deleteNotificationFunc](https://github.com/rook/rook/pull/9365/files#diff-39475f40154dad6ffbbe10347e21b251052adaf46f597105ad37cd8445ad65cfR102) has list of expected `notifications` for the `OB`. If that list does not match with OB then these functions return an error. So that reconcile function will return an error and that case is checked in each test case.